[WIP] Bump pytorch version

starride-teklia opened the pull request:
PyLaia currently requires an old version of Pytorch (1.6 - the latest version is 1.12). This is because PyLaia depends on [nnutils](https://github.com/jpuigcerver/nnutils), that supports only Pytorch 1.6. 

To bump the Pytorch version, we need to:

- release wheels for nnutils to support the latest Pytorch version (I have made a [contribution request](https://github.com/jpuigcerver/nnutils/issues/7) to do that)
- update the `requirements.txt` for PyLaia
- update the code and tests to make it compatible with Pytorch 1.12 (this should be easy, PyLaia does not depend on deprecated functions)

La0 reviewed (changes requested):
I get a failure on `tests/version_test.py` which seem related to these changes.

I currently have 46 failures (over ~1200), can you look into these too ? Most of them seem unrelated though.

starride-teklia commented:
Partially done:
* I fixed `tests/version_test.py` - this was due to an issue with `requirements.txt`
* I fixed some tests related to the pytorch upgrade

There are still 23 failed tests related to a multiprocessing issue - I will need more time to fix them. 
```py
23 failed, 1200 passed, 7 skipped, 4 xfailed, 156 warnings in 240.07s (0:04:00) 
```
We will also need to replace `torch.testing.assert_allclose` with `torch.testing.assert_close` as it will be [deprecated](https://github.com/pytorch/pytorch/issues/61844) soon.

La0 commented:
I simplified the CI to use only the dependencies mentionned in `requirements.txt`. But we still have some failing tests.

starride-teklia commented:
@La0 I fixed the CI for Python 3.8, 3.9 and 3.10
* I set `jsonargparse==4.7` as recent versions are not working properly (I will open an issue for that).           
* there was an issue related to matplotlib (`fig.canvas.set_window_title` => `fig.canvas.manager.set_window_title`)
* I removed some assertions checking the help message, as it changes a bit with Python > 3.8 (`type: Union[str, null], default: null` becomes `type: Optional[str], default: null`)

La0 commented:
All tests :ok:
